ractor_actors/examples/tcp.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) Sean Lawlor
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree.
+
+extern crate ractor_actors;
+
+use chrono::{DateTime, Utc};
+use ractor::{Actor, ActorProcessingErr, ActorRef};
+use ractor_actors::net::tcp::listener::*;
+use ractor_actors::net::tcp::session::*;
+use ractor_actors::net::tcp::*;
+use ractor_actors::watchdog;
+use ractor_actors::watchdog::TimeoutStrategy;
+use std::error::Error;
+use std::str::FromStr;
+use std::time::SystemTime;
+
+struct MyServer;
+
+struct MyServerState {}
+struct MyServerArgs {
+    /// Controls if the watchdog is used
+    watchdog: bool,
+
+    /// The port to listen on
+    port: NetworkPort,
+}
+
+#[cfg_attr(feature = "async-trait", async_trait::async_trait)]
+impl Actor for MyServer {
+    type Msg = ();
+    type State = MyServerState;
+    type Arguments = MyServerArgs;
+
+    async fn pre_start(
+        &self,
+        myself: ActorRef<Self::Msg>,
+        MyServerArgs { watchdog, port }: Self::Arguments,
+    ) -> Result<Self::State, ActorProcessingErr> {
+        let acceptor = MyServerSocketAcceptor { watchdog };
+
+        let _ = myself
+            .spawn_linked(
+                Some(format!("listener-{}", port)),
+                Listener::new(),
+                ListenerStartupArgs {
+                    port,
+                    encryption: IncomingEncryptionMode::Raw,
+                    acceptor,
+                },
+            )
+            .await?;
+
+        Ok(Self::State {})
+    }
+}
+
+struct MyServerSocketAcceptor {
+    watchdog: bool,
+}
+
+#[cfg_attr(feature = "async-trait", async_trait::async_trait)]
+impl SessionAcceptor for MyServerSocketAcceptor {
+    async fn new_session(&self, stream: NetworkStream) -> Result<(), ActorProcessingErr> {
+        tracing::info!("New connection: {}", stream.peer_addr());
+        Actor::spawn(
```

Comment:
these are spawned unlinked to the parent? They should be linked children of the server, so when the server shuts down all the children are terminated (i.e. network streams)

Reply:
Hm, perhaps. From the TCP/IP side, there is no problem keeping the client socket alive even if the listening socket has been closed. 

Anyway, it is up to the application given that it is implementing the `SessionAcceptor` trait and this example should be as ractor idiomatic as possible so I'll update it.